pick.lic

```diff
@@ -246,43 +246,56 @@ class LockPicker
       return
     end
 
-    if @make_pets
-      case bput("put my #{box} in my #{@settings.picking_pet_box_source}", 'You put your', "You just can\'t get", "There isn\'t any more")
-      when "You just can\'t get", "There isn\'t any more"
-        bput("put my #{box} in my #{@settings.picking_box_source}", 'You put your')
-        @pet_count = @pet_goal
-      when 'You put your'
-        @pet_count += 1
+    unless @settings.use_skeleton_key
+      if @make_pets
+        case bput("put my #{box} in my #{@settings.picking_pet_box_source}", 'You put your', "You just can\'t get", "There isn\'t any more")
+        when "You just can\'t get", "There isn\'t any more"
+          bput("put my #{box} in my #{@settings.picking_box_source}", 'You put your')
+          @pet_count = @pet_goal
+        when 'You put your'
+          @pet_count += 1
+        end
+        return
       end
-      return
-    end
 
-    if DRStats.thief?
-      if DRC.bput("glance my #{box}", /there.*(no|\d) traps? left/, 'I could').scan(/there.*(no|\d) traps? left/)[0].first.to_i > 0
-        DRC.message("Hit the Thief Hasten bug, disarming again!") if UserVars.lockpick_debug
-        attempt_open(box)
+      if DRStats.thief?
+        if DRC.bput("glance my #{box}", /there.*(no|\d) traps? left/, 'I could').scan(/there.*(no|\d) traps? left/)[0].first.to_i > 0
+          DRC.message("Hit the Thief Hasten bug, disarming again!") if UserVars.lockpick_debug
+          attempt_open(box)
+        else
+          attempt_pick(box)
+        end
       else
         attempt_pick(box)
       end
-    else
-      attempt_pick(box)
     end
 
     waitrt?
-    fput('stow left') unless @settings.use_lockpick_ring
+    stow_hands_except(box)
 
     waitrt?
     loot(box)
     dismantle(box)
     waitrt?
   end
 
+  # Put away lockpick or skeleton key, need both hands to loot and dismantle box.
+  def stow_hands_except(item)
+    fput('stow left') unless DRC.left_hand.nil? || DRCI.in_left_hand?(item)
+    fput('stow right') unless DRC.right_hand.nil? || DRCI.in_right_hand?(item)
+  end
```

Comment:
The lockpick or skeleton key or box could be in either hand. This logic stows whichever hand isn't holding the item to keep (e.g. the box)